### PR TITLE
Khkh/multiple installs checks

### PR DESCRIPTION
**Pull request by khkh-ms**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/VersionHelper.cs [hunk 1]
```diff
@@ -13,11 +17,26 @@ namespace Azure.Functions.Cli.Helpers
...
+            }
+
+            return olderVersionWarning;
+        }
 
         // Check that current core tools is the latest version. 
         // To ensure that it doesn't block other tasks. The HTTP Request timeout is only 1 second. 
         // We simply ingnore the exception if for any reason the check fails. 
-        public static async Task<string> CheckLatestVersionAsync()
+        public static async Task<(bool,string)> IsRunningAnOlderVersion()
```
Comment (kamperiadis) [inline src/Azure.Functions.Cli/Helpers/VersionHelper.cs:39]: Can we just make this function return Task<bool>? The only place where that message seems to be used anyways is for L33 of this file, so couldn't we just replace `olderVersionWarning` with `Constants.OldCoreToolsVersionMessage`?

**Review by kamperiadis — CHANGES_REQUESTED** [review]

File: src/Azure.Functions.Cli/Helpers/VersionHelper.cs [hunk 2]
```diff
@@ -13,11 +17,26 @@ namespace Azure.Functions.Cli.Helpers
 {
     internal class VersionHelper
     {
+        public static async Task<string> RunAsync(Task<(bool, string)> isRunningOlderVersion = null)
+        {
+            isRunningOlderVersion ??= IsRunningAnOlderVersion();
+
+            (var isOlderVersion, var olderVersionWarning) = await isRunningOlderVersion;
+
+            var multipleInstallsWarning = await HaveMultipleInstallations(isOlderVersion);
```
Comment (kamperiadis) [inline src/Azure.Functions.Cli/Helpers/VersionHelper.cs:26]: Couldn't we just have `HaveMultipleInstallations` not consume the `isOlderVersion` param and just have the `RunAsync` function handle the strings to be shown? That `isOlderVersion` param is only used to determine what message to present customers but it is not related to determining whether the customer has multiple installations.

Comment (khkh-ms) [reply src/Azure.Functions.Cli/Helpers/VersionHelper.cs:26]: I combined the old version and multiple versions message together when necessary. Showing both messages together was too wordy. 

I also started thinking the same as you are thinking but there is a part of the multiple version message being generated within `HaveMultipleInstallations` method. I wanted to generate the whole multiple version message at the same place therefore I decided to that in the method. The mentioned param was needed to help generate the full message as explained above.

File: src/Azure.Functions.Cli/Helpers/VersionHelper.cs (hunk 1 above)
Comment (khkh-ms) [reply src/Azure.Functions.Cli/Helpers/VersionHelper.cs:39]: Addressing it.

File: src/Azure.Functions.Cli/Helpers/VersionHelper.cs (hunk 2 above)
Comment (kamperiadis) [reply src/Azure.Functions.Cli/Helpers/VersionHelper.cs:26]: I think it is confusing for the naming convention of the function `HasMultipleInstallations` to serve two purposes (determine if there are multiple installations and then also determining what error message to provide users). In my opinion, it should only be used to determine if there are multiple installations and then the message handling should be determined elsewhere. If we need the directory information, then maybe it should not be named `HasMultipleInstallations` but rather something like `GetInstallationDirectories` and return the list of directories.

File: src/Azure.Functions.Cli/Helpers/VersionHelper.cs (hunk 1 above)
Comment (khkh-ms) [reply src/Azure.Functions.Cli/Helpers/VersionHelper.cs]: I have addressed it.

File: src/Azure.Functions.Cli/Helpers/VersionHelper.cs (hunk 2 above)
Comment (khkh-ms) [reply src/Azure.Functions.Cli/Helpers/VersionHelper.cs:26]: The output of the "where" command which determines the multiple installations, it is the same output which is used to generate the message. If we do the determine and message generation in the two different places, then we have run the run the "where" command twice which is already expensive. 

I agree with the name. Naming is sometimes tricky when we expected that the name should reflect the behavior the method when there is more going on the method. I am open for the suggestion.

File: src/Azure.Functions.Cli/Actions/HelpAction.cs [hunk 3]
```diff
@@ -97,18 +97,20 @@ public async override Task RunAsync()
                 DisplayGeneralHelp();
             }
 
-            await ShowLatestVersionMessage(latestVersionMessageTask);
+            // var isOldVersion = await latestVersionMessageTask; //await Constants.OldCoreToolsVersionMessage
```
Comment (kamperiadis) [inline src/Azure.Functions.Cli/Actions/HelpAction.cs:100]: LGTM! Just a minor comment. Can you remove this commented out code?

**Review by kamperiadis — APPROVED** [review]
LGTM!
